host.go

```diff
@@ -243,13 +243,22 @@ func (h *Host) Serve() {
 	h.listener.Serve()
 }
 
-func (h *Host) completeName(partial string) string {
+func (h *Host) completeName(partial string, currentUser *message.User) string {
 	names := h.NamesPrefix(partial)
 	if len(names) == 0 {
 		// Didn't find anything
 		return ""
 	}
 
+	for i, user := range names {
```

Comment:
Do we actually need a loop here, and a full array copy inside of it? We're just returning one name, could we check if the name we're returning is currentUser and if it is then return the one before that?

Also, then we don't need to pass in a `currentUser *message.User`, we could just pass in `skipName string` instead.

Reply:
thats sounds reasonable,yes.